packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -16,6 +16,70 @@ import '../build_configuration.dart';
 import '../globals.dart';
 import '../runner/flutter_command.dart';
 
+bool _addPackage(String directoryPath, List<String> dartFiles, Set<String> pubSpecDirectories) {
+  bool foundOne = false;
```

Comment:
This method could get a little simpler if you stored the length of dartFiles here, and compared that at the end to see if any files had been added.